test/test_load_ref_schema.rb

```diff
@@ -12,29 +12,19 @@ def load_other_schema
           }
         }
       },
-      Addressable::URI.parse("http://example.com/schema#")
+      Addressable::URI.parse(url)
     ))
   end
 
   def test_cached_schema
     schema_url = "http://example.com/schema#"
-    schema = {
-      "$ref" => schema_url
-    }
-    data = {}
-    load_other_schema
-    validator = JSON::Validator.new(schema, data)
+    load_other_schema(schema_url)
     assert JSON::Validator.schema_loaded?(schema_url)
```

Comment:
I believe the original intent here was to:

1. Add `http://json-schema.org/draft-04/schema#` to the `JSON::Validator::schemas` cache
2. Call `JSON::Validator.new(schema, data)` with a new schema that refers to `http://example.com/schema#` in a `$ref`
3. Check that creating a new validator adds the referenced schema to `JSON::Validator::schemas`